www/js/region-controller.js

```diff
@@ -17,14 +17,198 @@ angular.module('starter.controllers')
   RegionService.getRegion(residency).then(function(data) {
     // console.log("Retrieved region from service " + JSON.stringify(data));
     $scope.region=data;
+    console.log(JSON.stringify($scope.region));
   }, function(error) {
     console.log("Error retrieving region " + JSON.stringify(error));
   });
+  console.log(JSON.stringify($scope.region));
+  // $scope.offices=$scope.region.get("execOffAddrList");
+  // console.log(JSON.stringify($scope.offices));
+  // $scope.deleteErrorMessage=null;
+  $scope.delete=function(officeName){
+    $scope.offices=$scope.region.get("execOffAddrList");
+    console.log(JSON.stringify($scope.offices));
+    $scope.officeToBeDeleted=null;
+    for(var i=0;i<$scope.offices.length;i++)
+    {
+      if($scope.offices[i].officeName==officeName)
+      {
+        $scope.officeToBeDeletedIndex=i;
+        break;
+      }
+    }
+    $scope.offices.splice(i,1);
+    console.log(JSON.stringify($scope.offices));
+    for(var i=0; i < $scope.offices.length;i++)
+    { 
+      delete $scope.offices[i].$$hashKey;
+      console.log(JSON.stringify($scope.offices[i]));
+      if($scope.offices[i].execAdmin!=null) {
+        for(var j=0; j< $scope.offices[i].execAdmin.length;j++)
+        {
+          delete $scope.offices[i].execAdmin[j].$$hashKey;
+        }
+      }
+    }
+    console.log(JSON.stringify($scope.region));
+    console.log(JSON.stringify($scope.offices));
+    $scope.region.save(null, {
+      success: function(region) {
+        $scope.$apply(function(){
+              console.log("success");
+        });
+      },
+      error: function(region, error) {
+        console.log("Error in deleting the office " + error.message);
+        $scope.deleteErrorMessage="Unable to process your delete request.";
+      }
+    });
 
+  };
+  
   $scope.canUpdateRegion=AccountService.canUpdateRegion();
 
 })
 
+.controller('AddOfficeCtrl', function($scope, $stateParams, $state) {
+
+  var Region = Parse.Object.extend("Region");
+  var query = new Parse.Query(Region);
+  query.equalTo("uniqueName", $stateParams.regionUniqueName);
+  query.find({
+    success: function(regions) {
+      $scope.$apply(function(){
+        $scope.region=regions[0];
+         
+      });
+    },
+    error: function(error) {
+      console.log("Error retrieving region " + JSON.stringify(error));
+    }
+  });
```

Comment:
@pradeepKaatnam we should read region object from RegionService as much as possible to avoid additional network calls.